packages/popover/src/Popover.types.ts

```diff
@@ -70,65 +103,115 @@ export interface ChildrenFunctionParameters {
-       */
-      portalRef?: undefined;
-
-      /**
-       * When usePortal is `true`, specifies the scrollable element to position relative to.
-       */
-      scrollContainer?: null;
-    };
+/** @deprecated - use {@link RenderTopLayerProps} */
+export interface RenderInlineProps {
+  /**
+   * Popover element will render inline with the reference element
+   * @deprecated use 'top-layer'
+   */
+  renderMode?: 'inline';
+
+  /** Not used in this `renderMode` */
+  dismissMode?: never;
+
+  /** Not used in this `renderMode` */
+  onToggle?: never;
+
+  /** Not used in this `renderMode` */
+  portalClassName?: never;
+
+  /** Not used in this `renderMode` */
+  portalContainer?: never;
+
+  /** Not used in this `renderMode` */
+  portalRef?: never;
+
+  /** Not used in this `renderMode` */
+  scrollContainer?: never;
+
+  /** Not used in this `renderMode` */
+  usePortal: false;
+}
+
+/** @deprecated - use {@link RenderTopLayerProps} */
+export interface RenderPortalProps {
+  /**
+   * Popover element will render in a provided `portalContainer` or in a new div appended to the body
+   * @deprecated
+   */
+  renderMode?: 'portal';
+
+  /** Not used in this `renderMode` */
+  dismissMode?: never;
+
+  /** Not used in this `renderMode` */
+  onToggle?: never;
+
+  /**
+   * Specifies a class name to apply to the portal element
+   */
+  portalClassName?: string;
+
+  /**
+   * Specifies an element to portal within. If not provided, a div is generated at the end of the body
+   */
+  portalContainer?: HTMLElement | null;
+
+  /**
+   * Passes a ref to forward to the portal element
+   */
+  portalRef?: React.MutableRefObject<HTMLElement | null>;
+
+  /**
+   * Specifies the scrollable element to position relative to
+   */
+  scrollContainer?: HTMLElement | null;
+
+  /**
+   * Duplicated by `renderMode='portal'`
+   * @deprecated TODO: https://jira.mongodb.org/browse/LG-4526
+   */
+  usePortal?: true;
+}
+
+export interface RenderTopLayerProps {
+  /**
+   * Popover element will render in the top layer
+   */
+  renderMode?: 'top-layer';
+
+  /**
+   * Options to control how the popover element is dismissed
+   * - `'auto'` will automatically handle dismissal on backdrop click or key press, ensuring only one popover is visible at a time
+   * - `'manual'` will require that the consumer handle dismissal manually
+   */
+  dismissMode?: DismissMode;
+
+  /**
+   * A callback function that is called when the popover is toggled
+   */
+  onToggle?: (e: ToggleEvent) => void;
+
+  /** Not used in this `renderMode` */
+  portalClassName?: never;
+
+  /** Not used in this `renderMode` */
+  portalContainer?: never;
+
+  /** Not used in this `renderMode` */
+  portalRef?: never;
+
+  /** Not used in this `renderMode` */
+  scrollContainer?: never;
+
+  /** Not used in this `renderMode` */
+  usePortal?: never;
+}
+
+export type PopoverRenderModeProps =
+  | RenderInlineProps
+  | RenderPortalProps
+  | RenderTopLayerProps;
```

Comment:
I don't believe the `deprecated` flags above will be seen by consumers, since we're grouping them into this type here